x/vote/types/types.go

```diff
@@ -45,3 +46,16 @@ func (m *TalliedVote) TallyVote(voter sdk.ValAddress, votingPower sdk.Uint, isLa
 	m.IsVoterLate[voter.String()] = isLate
 	m.Tally = m.Tally.Add(votingPower)
 }
+
+// ValidateBasic returns an error if the TalliedVote is not valid
+func (m TalliedVote) ValidateBasic() error {
+	if m.Data == nil {
+		return errors.New("data is nil")
+	}
+
+	if m.Tally.IsZero() {
+		return errors.New("vote tally is zero")
+	}
+
```

Comment:
Also keys in the map of `.IsVoterLate` are all valid string representations of val addresses?